internal/crypto/src/time_stamp/response.rs

```diff
@@ -0,0 +1,192 @@
+// Copyright 2022 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License,
+// Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+// or the MIT license (http://opensource.org/licenses/MIT),
+// at your option.
+
+// Unless required by applicable law or agreed to in writing,
+// this software is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR REPRESENTATIONS OF ANY KIND, either express or
+// implied. See the LICENSE-MIT and LICENSE-APACHE files for the
+// specific language governing permissions and limitations under
+// each license.
+
+use asn1_rs::nom::AsBytes;
+use bcder::decode::Constructed;
+use rasn::{AsnType, Decode, Encode};
+
+use crate::{
+    asn1::{
+        rfc3161::{PkiStatus, TimeStampResp, TimeStampToken, TstInfo, OID_CONTENT_TYPE_TST_INFO},
+        rfc5652::{SignedData, OID_ID_SIGNED_DATA},
+    },
+    time_stamp::TimeStampError,
+};
+
+/// TO REVIEW: Does this need to be public?
+pub struct TimeStampResponse(pub TimeStampResp);
+
+impl std::ops::Deref for TimeStampResponse {
+    type Target = TimeStampResp;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl TimeStampResponse {
+    /// Convert a DER-encoded `TimeStampRes` structure to a `TimeStampResponse`
+    /// struct.
+    ///
+    /// TO REVIEW: Does this need to be public after refactoring?
+    pub fn parse_der(ts_response: &[u8]) -> Result<Self, TimeStampError> {
+        Ok(Self(
+            Constructed::decode(ts_response, bcder::Mode::Der, |cons| {
+                TimeStampResp::take_from(cons)
+            })
+            .map_err(|e| TimeStampError::DecodeError(e.to_string()))?,
+        ))
+    }
+
+    /// Return `true` if the request was successful.
+    pub fn is_success(&self) -> bool {
+        matches!(
+            self.0.status.status,
+            PkiStatus::Granted | PkiStatus::GrantedWithMods
+        )
+    }
+
+    /// TO REVIEW: Does this need to be public?
```

Comment:
I don't think any of these need to be public.  They may be needed by verify.rs however.

Reply:
Good catch. Downgraded to `pub(crate)` and dead-stripped.